tests/scripts/all.sh

```diff
@@ -3818,6 +3818,61 @@ component_test_psa_crypto_config_reference_cipher_aead () {
     tests/compat.sh -V NO -p mbedTLS
 }
 
+component_test_full_block_cipher_psa_dispatch () {
+    msg "build: full + PSA dispatch in block_cipher"
+
+    loc_accel_list="ALG_ECB_NO_PADDING \
+                    KEY_TYPE_AES KEY_TYPE_ARIA KEY_TYPE_CAMELLIA"
+
+    # Configure
+    # ---------
+
+    # Start from the full config
+    helper_libtestdriver1_adjust_config "full"
+
+    # Disable CCM/GCM support in PSA as we are testing dispatching of legacy
+    # modules (GCM_C/CCM_C through BLOCK_CIPHER_C) to PSA.
+    scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CCM
+    scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_GCM
+
+    # Disable CIPHER_C because we want legacy GCM_C/CCM_C to use BLOCK_CIPHER_C.
+    scripts/config.py unset MBEDTLS_CIPHER_C
```

Comment:
Note: that's a usability problem. It will result in annoying documentation/ChangeLog: "Built-in GCM and CCM can now take advantage of a plain AES driver, but only if CIPHER_C is disabled. If you want those to use your driver but still need CIPHER_C for some reason (like using PKCS5/12, NIST-KW, CMAC, encrypted PEM) then tough luck."

I wonder if we should change the logic to:
- `GCM_C` & `CCM_C` will use `BLOCK_CIPHER_C` as soon as it's enabled;
- we auto-enable it if (`GCM_C` or `CCM_C` is enabled) and (`CIPHER_C` is disabled or `MBEDTLS_BLOCK_CIPHER_SOME_PSA` is defined). (Of course make the definition of `MBEDTLS_BLOCK_CIPHER_SOME_PSA` no longer conditional on `BLOCK_CIPHER_C` being enabled in the first place or we have a loop.)

Wdyt? (Could be a follow-up.)

Reply:
The main consequence I see here is that `BLOCK_CIPHER_C` and `CIPHER_C` are no more mutually exclusive: in a build which has `CIPHER_C` and there is a PSA driver available for AES keys, we end up having an increased code size. The advantage is that driver might be faster/safer for AES computations though.

Reply:
Commits from https://github.com/Mbed-TLS/mbedtls/pull/8632/commits/8f8a598c77d71fb71da45bd61bcc1f24a6b5619d to https://github.com/Mbed-TLS/mbedtls/pull/8632/commits/16fa1a7577fa086a3306cb44b4d0b7497cdb5a19 implement this.